libraries/utilities/plugin_code_generator.py

```diff
@@ -0,0 +1,138 @@
+# The working directory of the script is the project root directory.            #
+# Script is called from libraries/utilities/CMakeLists.txt                      #
+#################################################################################
+
+
+import json 
+import os
+
+
+# list of files where generated code is used (relative to project root)
+# libraries/app/include/graphene/api.hpp
+# libraries/app/api.cpp
+# libraries/app/application.cpp
+
+generated_files_root_dir = "libraries/utilities/include/graphene/utilities/"
+
+plugin_root_dir = "libraries/plugins/"
+
+generated_code_warning = "\
+/*****************************************/\n\
+/* This code is generated automatically. */\n\
+/* To generate new plugin code create a  */\n\
+/* plugin.json file in a subdirectory    */\n\
+/* of the plugin root directory.         */\n\
+/*****************************************/\n\
+\n\
+\n"
+
+# this is a dictionary of the generated filename and the generated code
+# variables in the code are replaced by the variables from the plugin.json file
+templates = {
+"generated_api_cpp_case_code.hpp":
+"else if( api_name == \"{plugin_name}_api\" ) {{ \n\
+    _{plugin_name}_api = std::make_shared< {plugin_name}_api >( std::ref(_app) );\n\
+}} \n",
+
+"generated_api_cpp_login_api_code.hpp":
+"fc::api<graphene::{plugin_namespace}::{plugin_name}_api> login_api::{plugin_name}() const \n\
+{{ \n\
+    FC_ASSERT(_{plugin_name}_api); \n\
+    return *_{plugin_name}_api; \n\
+}}\n\n",
+
+"generated_api_hpp_include_code.hpp":
+"#include <graphene/{plugin_name}/{plugin_name}_api.hpp> \n",
+
+"generated_api_hpp_login_api_method_def_code.hpp":
+"/// @brief Retrieve the {plugin_name} API \n\
+fc::api<graphene::{plugin_namespace}::{plugin_name}_api> {plugin_name}()const;\n",
+
+"generated_api_hpp_login_api_apiobj_code.hpp":
+"optional< fc::api<graphene::{plugin_namespace}::{plugin_name}_api> > _{plugin_name}_api;\n",
+
+"generated_api_hpp_reflect_code.hpp":
+"({plugin_name})\n",
+
+"generated_application_cpp_allowed_apis_code.hpp":
+"wild_access.allowed_apis.push_back( \"{plugin_name}_api\" );\n",
+
+}
+
+
+# searches for all plugin.json files in the plugin directory
+# parses them and returns a list of dictionarys with the parsed data
+def get_all_plugin_data():
+    
+    # holds content of all plugin.json files as dictionary
+    plugin_data_list = []
+    # traverse plugin root dir search for plugin.json files
+    for root, dirs, files in os.walk( "./" + plugin_root_dir ):
+        for file in files:
+            if file == "plugin.json":
+                with open( os.path.join( root, file ) ) as f:
+                    parsed_data = json.load( f )
+                    data_dic = dict()
+                    data_dic['plugin_name']      = parsed_data['plugin_name']
+                    data_dic['plugin_namespace'] = parsed_data['plugin_namespace']
+                    data_dic['plugin_project']   = parsed_data['plugin_project']
```

Comment:
I'm not a python expert - is it really necessary to copy the fields instead of appending `parsed_data` directly?